activities/Gwen.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+import os
+
+import grass.script as gs
+
+
+def run_function_with_points(scanned_elev, env, points=None, **kwargs):
+    """Doesn't do anything, except loading points from a vector map to Python
+
+    If *points* is provided, the function assumes it is name of an existing vector map.
+    This is used during testing.
+    If *points* is not provided, the function assumes it runs in Tangible Landscape.
+    """
+    if not points:
+        # If there are no points, ask Tangible Landscape to generate points from
+        # a change in the surface.
+        points = "points"
+        import analyses
+
+        analyses.change_detection(
+            "scan_saved",
+            scanned_elev,
+            points,
+            height_threshold=[10, 100],
+            cells_threshold=[5, 50],
+            add=True,
+            max_detected=5,
+            debug=True,
+            env=env,
+        )
+
+        # generate random point for focal garden
+        gs.run_command("v.random", output="focal_garden", npoints=1, seed=3)
+
+        # calculate connectivity between focal garden and points from pins
+        gs.run_command(
+            "r.cost",
+            input="cfactorgrow_1m",
+            start_points="focal_garden",
+            stop_points="points",
```

Comment:
You want `points` variable here.
```suggestion
            stop_points=points,
```